src/jit/flowgraph.cpp

```diff
@@ -7056,6 +7056,19 @@ GenTreeCall* Compiler::fgGetStaticsCCtorHelper(CORINFO_CLASS_HANDLE cls, CorInfo
     GenTreeCall* result = gtNewHelperCallNode(helper, type, argList);
     result->gtFlags |= callFlags;
 
+    // If we're jitting the special EqualityComparer<T>.Default
+    // intrinsic, we can remove the shared helper call if the
+    // associated field lookup is unused.
+    if ((info.compFlags & CORINFO_FLG_JIT_INTRINSIC) != 0)
```

Comment:
Minor suggestion: you might want to say that we are marking it so that later, *if* it is unused, we can remove it. Otherwise it almost reads as if the code here should be doing the removal.

Reply:
Fixed.